pkg/storaged/btrfs/btrfs-tool.py

```diff
@@ -241,15 +231,39 @@ def cmd_poll(opt_mount):
     sys.stdout.flush()
 
 
+def unshare_mounts():
+    # The os.unshare function is available since Python 3.12, but we
+    # still need to support older Pythons.
+    if "unshare" in os.__dict__:
+        os.unshare(os.CLONE_NEWNS)
+    else:
+        libc = ctypes.CDLL(None)
+        libc.unshare.argtypes = [ctypes.c_int]
+        get_errno_loc = libc.__errno_location
+        get_errno_loc.restype = ctypes.POINTER(ctypes.c_int)
+        ret = libc.unshare(2**17)
+        if ret != 0:
+            errno = get_errno_loc()[0]
+            raise OSError(errno, os.strerror(errno))
+
+
 def cmd_do(uuid, cmd):
     debug(f"DO {uuid} {cmd}")
-    with mount_database() as db:
-        filesystems = list_filesystems()
-        for fs in filesystems.values():
-            if fs['uuid'] == uuid:
-                mp = force_mount_point(db, fs, opt_repair=True)
-                with context_chdir(mp):
-                    subprocess.check_call(cmd)
+    filesystems = list_filesystems()
+    for fs in filesystems.values():
+        if fs['uuid'] == uuid:
+            path = tempfile.mkdtemp()
+            dev = fs['devices'][0]
+            try:
+                unshare_mounts()
+                subprocess.check_call(["mount", "--make-rprivate", "/"])
+                subprocess.check_call(["mount", dev, path])
+                try:
+                    subprocess.check_call(cmd, cwd=path)
+                finally:
+                    subprocess.check_call(["umount", path])
```

Comment:
This isn't a long-lived process, right? So as the `umount` is not reliable, we may just leave it out completely. Exiting btrfs-tool.py ought to clean it up on the kernel side.

Reply:
Yes, but we also want to remove the `path` temporary directory. In order to do that, we need to unmount first.

Reply:
Hence my suggestion to just use `/run/cockpit/btrfs`. Then you don't need to clean it up, then you don't need to umount it, and a lot of complexity here just dissolves into nothingness.

Reply:
> Hence my suggestion to just use `/run/cockpit/btrfs`

Oh, I missed that! Can we use that? I guess we can! Excellent!

Reply:
Well, you need to `mkdir -p` it, but I don't see why not?

Reply:
Done.